src/gpu/locks.rs

```diff
@@ -201,9 +212,9 @@ macro_rules! locked_kernel {
                 }
             }
 
-            pub fn with<F, R>(&mut self, mut f: F) -> GpuResult<R>
+            pub fn with<Fun, R>(&mut self, mut f: Fun) -> GpuResult<R>
```

Comment:
A documentation comment would be useful here, since this is pub function and the private functions above are already documented.  Looks like this is provided with an initialization function to be called after GPU is initialized?